crypto/src/main/java/net/consensys/cava/crypto/mikuli/BLS12381.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package net.consensys.cava.crypto.mikuli;
+
+import net.consensys.cava.crypto.mikuli.group.AtePairing;
+import net.consensys.cava.crypto.mikuli.group.G1Point;
+import net.consensys.cava.crypto.mikuli.group.G2Point;
+import net.consensys.cava.crypto.mikuli.group.GTPoint;
+
+import java.util.List;
+
+import org.apache.milagro.amcl.BLS381.BIG;
+import org.apache.milagro.amcl.BLS381.ECP;
+import org.apache.milagro.amcl.BLS381.ECP2;
+import org.apache.milagro.amcl.BLS381.MPIN;
+
+public final class BLS12381 {
+
+  /**
+   * Generates a SignatureAndPublicKey.
+   *
+   * @param keyPair The public and private key pair, not null
+   * @param message The message to sign, not null
+   * @return The SignatureAndPublicKey, not null
+   */
+  public static SignatureAndPublicKey sign(KeyPair keyPair, byte[] message) {
```

Comment:
can we replace byte[] with Bytes?

Reply:
Milagro wants bytes[].  are you saying that you want me to make it a Bytes type and convert it back and forth each time it interacts with Milagro?

Reply:
Personally, I'd do an overloaded method.